x/evm/types/command.go

```diff
@@ -89,9 +89,9 @@ func NewDeployTokenCommand(chainID sdk.Int, keyID multisig.KeyID, asset string,
 }
 
 // NewMintTokenCommand creates a command to mint token to the given address
-func NewMintTokenCommand(keyID multisig.KeyID, id CommandID, symbol string, address common.Address, amount *big.Int) Command {
+func NewMintTokenCommand(chainID sdk.Int, keyID multisig.KeyID, id nexus.TransferID, symbol string, address common.Address, amount *big.Int) Command {
 	return Command{
-		ID:         id,
+		ID:         NewCommandID(id.Bytes(), chainID),
```

Comment:
Need to think a bit about changing how command ID is derived, might lead to collisions. But the chain ID domain separation isn't necessary since transfer ID is a global counter.

This will also break axelarscan as it's currently done at least. For historical reasons, they assume that commandID is hex encoding of the transferID. I'll have to see how they index it exactly but they haven't moved to using the MintCommand event indexing.

Reply:
non of the other commands use transfer ID though. So I simply used the same mechanism as the others.
There can be just as many collisions as before (and we check for collisions)

Reply:
I don't mean collisions in terms of pure randomness. But constructing collisions because inputs can be forced to be the same, i'll have to take a look

Reply:
I don't see how. It's still taking the transferID as input, so as long as we don't have a random hash collision, I don't think there is a way to force a collision